src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -588,14 +589,24 @@ class CornerstoneViewport extends Component {
   };
 
   onNewImage = event => {
-    const newImageId = event.detail.image.imageId;
-    const newImageIdIndex = this.props.imageIds.indexOf(newImageId);
+    const { imageId } = event.detail.image;
+    const { sopInstanceUid } = cornerstone.metaData.get(
```

Comment:
It's possible for a `cornerstone.metaData.get` to return `undefined`. That would cause this to throw an exception. You could do:

```js
const { sopInstanceUid } = cornerstone.metaData.get('generalImageModule', imageId) || {};
```

Reply:
Sure thing. Updated.